### PR TITLE
Torp (and volatility component fixes, again, this time for real)

### PR DESCRIPTION
**Pull request by DeltaFire15**
## About The Pull Request
First of all, fixes volatility. The component (for taking damage) relied upon a signal that.. nothing except overmaps sent. So, they never triggered, from damage at least. This has been rectified and any /obj now sends the signal (overmaps slightly adjusted to not doublesend it)

Secondly, players now get access to a slightly weaker version of the disruption torpedo (found on the Aspala class stealth sub) which has lead to quite some "fun" with its EMP capabilities, thanks to some reverse engineering efforts from whiterapids (or well, if whiterapids were actually done finishing their designs, oh well, how convenient the ship has a research department isn't it?)
These have a special effect against non-player ships <sub>(okay, player fighters also get this shhh)</sub> which causes stacking probability for a ship to effective be stunned for a processing tick. They are expensive and have a lower disruption cap than the AI version, however they're nonetheless a very effective support ammunition type, especially against enemies capable of shaking off a lot of pure damage. they also may prove to be quite an asset in PVP.
Sprites for the warhead and ammunition itself by Pontenerd.
<sup><sub>Handle with care, Whiterapids is not responsible for injury, malfunctions, disruption of IPC vitals, premature detonations, or any combinations of the previous. Consult the technical hotline for further details.</sub></sup>

Additionally, Hellfire torps have had a little rewrite to make them more of a threat to non-z-level possessing ships. They now apply hullburn to enemy ai ships <sub>(and player fighters, be careful!)</sub> which deals damage over time. Damage for the player version has been reduced to 300 (from 400), but it now deals a total of ~~600~~ 360 additional damage over 60 total processing ticks, coming together in a hefty ~~900~~ 660(!) raw damage. (Hullburn reduced because I forgor we nerfed ai hp a long while ago)
Stacking these is only semi-effective, for hullburn, only the strongest DoT value at a time is used, however, additional applications extend the queued time for the strongest hullburn effect currently active. So, e.g. Applying a hullburn with 360 damage over 60 ship ticks and then another with 100 damage over 60 ship ticks would not deal 460 over 60 ship ticks, but instead 720 damage over 120 seconds as the time of the stronger hullburn previously inflicting 360 damage total is effectively doubled. (The way stacking hits / hullburn durations works is probably subject to change)
This makes them a very powerful tool if used against ships unable to repair, allowing for hit-and-run attacks that can chip away at even heavy hulls (shields do not block hullburn damage, it is however mitigated by fire armor which no ship currently has but can be added for those meant to resist hullburn). Against ships with z levels such as in pvp, they still incinerate interior sections as before.
<sub><sup>Major fire hazard, store only in class III containment units. Insurance is void for any injuries or damage caused by runaway reactions occurring due to complications from improper storage.</sup></sub>

## Why It's Good For The Game
More torp. More good. Make torp.
Although to be more serious, having more variety of ordnance available to players may allow some new strategies, especially related to the commonly viewed as "too weak" torpedo weapon class.
Volatility working again is also just a very good thing.

## Changelog
:cl: DeltaFire15, Pontenerd
fix: Volatility components should properly react to damage again.
add: Reverse engineering of the feared disruption torpedo has almost reached a breakthrough. Whiterapids is willing to transmit schematics of the research progress to any military vessels equipped with onboard research facilities.
balance: Hellfire torpedoes when used against AI vessels now inflict hullburn, a damage over time effect melting away at a ship's hull. Player accessible torp impact damage lowered to 300 from 400. Damage over time added with 360 total damage over 60 ship processing ticks, amounting to a total 660 raw damage.
/:cl:

### CODE REVIEW

**Comment by Karmic-Skink** [comment]
If you wanted to simulate that hellburn effect on our ships properly.

Add an impacted torp shell object to the initial plasma burst location and have it emit waves of plasma fire for a short time.

**Comment by DeltaFire15** [comment]
> If you wanted to simulate that hellburn effect on our ships properly.
> 
> Add an impacted torp shell object to the initial plasma burst location and have it emit waves of plasma fire for a short time.

Well, it's more of the opposite, the hullburn effect emulates the interior fire causing damage since ai ships don't have "z damage"

**Comment by Karmic-Skink** [comment]
> > If you wanted to simulate that hellburn effect on our ships properly.
> > Add an impacted torp shell object to the initial plasma burst location and have it emit waves of plasma fire for a short time.
> 
> Well, it's more of the opposite, the hullburn effect emulates the interior fire causing damage since ai ships don't have "z damage"

Well in most cases the single room fires generated don't do much, so maybe tone the DoT down? *shrug*

File: nsv13/code/modules/munitions/ship_weapons/volatility_component.dm [hunk 1]
```diff
@@ -59,3 +61,34 @@ Add this component to an atom to mark it as volatile, if it takes fire damage, i
...
+	if(!parent)
+		message_admins("Volatility component tried to explode with no attached parent. Contact a coder")
+		return FALSE
+	log_game("Volatile substance caused a runaway hellfire reaction at [get_area(parent)]")
+	var/base_ex_power = explosion_scale * volatility
+	explosion(parent, 0, 0, base_ex_power, base_ex_power * 2, base_ex_power)
+	var/turf/detonation_turf = get_turf(parent)
+	detonation_turf.atmos_spawn_air("o2=20;plasma=110;TEMP=700")
+	if(!QDELETED(parent))
+		qdel(parent)
```
Comment (TheFakeElon) [inline nsv13/code/modules/munitions/ship_weapons/volatility_component.dm:94]: smh this is not how you're supposed to use components. Components should be generic, special implementation should always be done on the object

Comment (DeltaFire15) [reply nsv13/code/modules/munitions/ship_weapons/volatility_component.dm:94]: I mean this is kinda what oop exists for no?
Is there any actual practical disadvantage, considering subtyping these prevents having to copypaste behaviors for projectiles with simillar volatility reactions but different inheritance trees?

**Comment by Absolutely-Steph** [comment]
Now that the volatility thing works, I'm noticing that the explosions it generates are incredibly puny. gunpowder does like a 0,0,1,1 explosion which isn't even enough to detonate other gunpowder on the same tile as itself.
I exploded a nuke torp and it didn't even breach the hull, it just mildly scratched some plating around itself and that's it.

**Comment by DeltaFire15** [comment]
> Now that the volatility thing works, I'm noticing that the explosions it generates are incredibly puny. gunpowder does like a 0,0,1,1 explosion which isn't even enough to detonate other gunpowder on the same tile as itself. 

Yes, normal gunpowder is comperatively low on the volatility scale as opposed to torpedoes or plasma accelerant, which also means it is less likely to chain react if clipped by other volatiles' explosions.

>I exploded a nuke torp and it didn't even breach the hull, it just mildly scratched some plating around itself and that's it.

Nukes don't exist anymore. If you mean hellfire torps, those cause a fire with ~25% of the severity of a "true" hellfire hit in their room. Which, where it was tested likely was exposed to space considering it'd have been noticed otherwise.

**Comment by Karmic-Skink** [comment]
Nukes were confiscated ages ago, also they wouldn't reach criticality from external means.

File: nsv13/code/modules/munitions/ship_weapons/volatility_component.dm (hunk 1 above)
Comment (TheFakeElon) [reply nsv13/code/modules/munitions/ship_weapons/volatility_component.dm:94]: Components are ECS, not OOP

**Review by Bokkiewokkie — APPROVED** [review]

**Pull request merged**